### ISSUE TITLE
running mako twice using the same tree is problematic when changing file.type

### ISSUE THREAD

**Issue by matthewmueller** [makojs/core#4]
the following code is problematic:

```
var mako = Mako()
  .use(jade())

mako
  .build('index.jade')
  .then(noop)

file = mako.tree.getFile('index.jade')
file.analyzed = false

mako
  .build('index.jade')
  .then(noop)
```

because the `file.type` gets changed from `jade` to `html` in the jade plugin and the hooks are dependent on `file.type`. so after you compile it once, the `file.type` has changed and so it won't be recompiled when that file changes again. this makes doing file watchers pretty difficult. 

could be thinking about this in the wrong way though.

**Comment by matthewmueller** [comment]
ah, was able to solve this by resetting the `file.type` in the watcher. i think it's a bit of a hack, but should work okay.

**Comment by dominicbarnes** [comment]
Ah dang, I'll need to think about that some more. I may need to reset any changed `file.type` before re-running `analyze()`.

**Comment by matthewmueller** [comment]
yah maybe introduce and output type or something. might even make sense to create a new node in the tree, not sure.

**Comment by dominicbarnes** [comment]
I believe I've resolved this issue @matthewmueller

In mako-tree@0.5, I've added `File#dirty()`, which can be used by plugins (eg: [mako-stat](https://github.com/makojs/stat/blob/master/index.js#L21)) to mark files so they will be analyzed again.

In mako-core@0.3, the original file type will always drive the `preread` hook, giving it the opportunity to be marked dirty. (which also resets the file type) I believe this should cover your situation quite nicely, let me know if it does not.
